### ISSUE TITLE
Sol Trigger  the talking's volume are too low and missing voices if skipping cutscene

### ISSUE THREAD

**Issue by daniel229** [hrydgard/ppsspp#6511]
DEMO has the same problem
DEMO download
https://drive.google.com/file/d/0BzGZGDfFE68zRXF5Y0dSUUZfUXc/edit?usp=sharing

DEMO load the savedata and go up.(rename jpg to rar)
![npjh90291data00](https://cloud.githubusercontent.com/assets/3481559/3488520/4c221e2e-04ec-11e4-9bd8-ba30ccd7317a.jpg)

the videoes (rename jpg to mp4)
![normal](https://cloud.githubusercontent.com/assets/3481559/3488529/c0419492-04ec-11e4-832f-8f851cb75377.jpg)
![missing voice](https://cloud.githubusercontent.com/assets/3481559/3488530/c50edcb4-04ec-11e4-85a7-88b8b5fe15c4.jpg)

**Comment by unknownbrackets** [comment]
Hmm, currently the voices don't sound obviously wrong compared to the music.  Is this still a problem?

-[Unknown]

**Comment by daniel229** [comment]
Still missing voices anafter skipping cutscene. and the volume still too low compare to PSP.

**Comment by unknownbrackets** [comment]
But the voices are already louder than the music by a good margin?  There's one guy who talks really quiet, though.  I'll try it later on a PSP.

-[Unknown]

**Comment by daniel229** [comment]
Yeah,in game config low down the BGM and max up the SE&Voice,then turn up the computer speaker from 10% to 50%.then the voice and BGM seems good.

**Comment by unknownbrackets** [comment]
Hmm, I just had default settings.  Is it only in certain areas?

-[Unknown]

**Comment by daniel229** [comment]
No,I just said low down the BGM volume to make it close to the voice volume level,feel better.

**Comment by unknownbrackets** [comment]
So here's what I did:

1. Start demo fresh.
2. Select new game.
3. Listen to first few seconds of video (audio seems fine, very audible), and press start to skip.
4. Listen as radio noise comes through (first text dialog.)  Sounds not entirely clear, radio-like.
5. Scarface guy talks.  This guys a bit quiet, though I can make out what he's saying.
6. Purple girl talks.  She sounds fine, a bit louder than the background music, pretty much where I'd want voices.
7. Scarface talks again.  Mr. Quiet, he sounds a bit far away, but I can still hear him as before.
8. Black hair guy talks.  His voice is audible but it's hard for me to hear clearly.
9. Screen zooms in, radio plays.  Sounds fine, much clearer this time.

So next, playing it on my PSP with my headset.

1. Video sounds the same.
2. Radio sounds possibly a bit clearer, but not significantly different.
3. Mr. Quiet is still Mr. Quiet.
4. Purple still has an understandable voice.
5. Mr. Quiet went a little far away again when he talked.
6. Still think Blacky's voice is hard to understand, though he's loud enough.
7. Radio similarly clearer on the second round as before.
8. Mr. Quiet at least agrees loudly enough, although then he runs away again for the rest of his dialog.

So, for me, it seems exactly the same.  Music sounds relatively the same in both cases.  I wonder if there's some setting or something involved here?

-[Unknown]

**Comment by daniel229** [comment]
Is it the BGM louder than the PSP,turn up the computer volume to make Mr.Quiet sound clearly enough,the BGM would be pretty louder.
Low down volume of BGM like this it sound pretty like the PSP.
![02](https://cloud.githubusercontent.com/assets/3481559/6429519/97724458-c012-11e4-869b-7c1faa75ba2a.png)

**Comment by unknownbrackets** [comment]
Everything sounded the same to me on my desktop and the PSP.

I sped through everything and the settings are:
BGM = middle
SE & Voice = full

AFAICT this is the default, right?

An example, right after the new game, while I'm free to walk for the first time, if I walk toward the camera, Purple says something to me.  You can hear the background music go quieter, although she's not very loud there.

During the cutscene right after that, Mr. Quiet is not so quiet, but it seems like that's just the game...

Anyway, I don't have the best ears, maybe I'm just not hearing the difference.

-[Unknown]

**Comment by vsub** [comment]
Well I just tried the game(on the latest beta)with my save at the end of the game and once I skip all of the cut scenes(and the battle with the final boss started)almost all of the voice overs and sound effects were gone.

Also since there are couple of voice over scenes one after another,if you skip one,all of the rest will not have voice overs...and music too(no sound at all)

**Comment by daniel229** [comment]
Some voice is worse, like these savestate(or using the top post savedata,load it and go up the street),possibly device problem.
http://1drv.ms/1LZUXF5

”チッ”　seems clearly on PSP.

**Comment by unknownbrackets** [comment]
Grey hair up guy must be Mr. Quiet's twin brother.

When I load the save, or get there from the beginning, the first thing I notice is that Sophie's skill (shooting bullets) is quiet.  I'm not trying this on a PSP but it sounds obviously wrong.  There's also no sound when they finish the battle which they move their mouths, so looks wrong also.

So there's definitely some problem with silent audio... or are you saying the talking here would be audible with the adjusted settings?

-[Unknown]

**Comment by daniel229** [comment]
I am talking the audible talking,and that silent voice is usually happen after skipping the cutscene.

**Comment by unknownbrackets** [comment]
So the issue here with voices not playing after skipping the cutscene is that skipping pauses the voices, and nothing unpauses.  It might be that __sceSasSetVoiceATRAC3 or __sceSasUnsetATRAC3 should unpause or return an error if the voice is paused, will have to test.

-[Unknown]

**Comment by daniel229** [comment]
Fixed,the voices difference is ECHO or other thing.

**Comment by unknownbrackets** [comment]
Ah, so missing reverb was why it was quieter?  Well, I have terrible ears, but glad it's better now.

-[Unknown]

**Comment by daniel229** [comment]
Yes,sound louder,better now,more like on PSP.

**Comment by unknownbrackets** [comment]
So, right now the behavior is a bit wrong.  I don't know if it'll cause problems now, but it would when we switch to using PSP RAM for the buffer.

So, when you setup SAS with ATRAC3, and play, it will eventually set the `second.addr` pointer to -1 or 0, or similar (still need more tests, this part is based on game behavior), this tells the game it needs to send more data, which it polls this value.  Since we always set it to 0, it always thinks more data is needed, and sends more, overwriting the old data.

Still need to work out the correct semantics for when more data is needed, but I found that asking for it ~12 frames ahead made the game still play correctly, and made it load the audio data much more evenly.  Using ~4 frames, it failed decoding at some point because the game didn't provide additional data early enough.

-[Unknown]

**Comment by unknownbrackets** [comment]
I've confirmed via tests the -1 when it runs low on data, but haven't worked out the exact semantics yet.  It seems like when the buffer is half empty, perhaps.

This could definitely be causing issues, though.  I've noted that if I return values wrongly, voices can get cut off, silent, or etc.  I wonder if the improper behavior could even be confusing God Eater or other games using this feature.

-[Unknown]

**Comment by IrfanH495** [comment]
sorry, I've experienced that in v1.17.1-150-
https://youtu.be/x8cnIlp9aXo
Should I create a new issue or wait for this to open.?

**Comment by IrfanH495** [comment]
@hrydgard 
v1.16.6 The character's voice is still there.
v1.17 Character voices disappear in cutscenes

**Comment by hrydgard** [comment]
Hm, so a regression. The [new buildbot](https://www.ppsspp.org/devbuilds/) has all the builds in that range now - could you try to narrow it down a little?

**Comment by IrfanH495** [comment]
@hrydgard 
I was confused about the file name on my device
![20240407_074918](https://github.com/hrydgard/ppsspp/assets/78854859/9fe0fa02-2920-40bf-942c-b1d6de3949c2)

I found it
v1.16.6-310 there are still character voices
v1.16.6-312 the character's voice is missing

**Comment by hrydgard** [comment]
Hm, that's a thought - now that I have control over the buildbot, I should make the build APKs named after the versions...

Alright, so this is another one of the atrac3 loop problems: it's @sum2012 's https://github.com/hrydgard/ppsspp/pull/18414

**Comment by hrydgard** [comment]
Should work since #20138 (though really fixed by #20098). 

Though hasn't been extensively tested. Please re-open if there are still issues.

**Comment by hrydgard** [comment]
#20156 should fix this.
